NavigationReactNative/src/android/src/main/java/com/navigation/reactnative/TabNavigationManager.java

```diff
@@ -21,6 +24,13 @@ public void setBottomTabs(TabNavigationView view, boolean bottomTabs) {
         view.bottomTabs = bottomTabs;
     }
 
+    @ReactProp(name = "labelVisibility")
+    public void setLabelVisibilityMode (TabNavigationView view, int labelVisibilityMode) {
+        view.labelVisibility = labelVisibilityMode;
+        view.setVisibility();
+    }
```

Comment:
Here's the twitter sample with label visibility set to "auto"
![Screenshot 2022-02-11 at 17 11 37](https://user-images.githubusercontent.com/1761227/153637059-59254a15-0427-44eb-a6ac-f28c45af95d2.png)

And here it is with "selected"
![Screenshot 2022-02-11 at 17 10 20](https://user-images.githubusercontent.com/1761227/153636948-edf22ca9-1881-4989-9410-74d079e34ca4.png)

You can see that Android changes the width of the tabs when label visibility is "selected". I think when there are more than 3 tabs this width change isn't noticeable. But now we're allowing "selected" with <= 3 tabs the width change isn't good.

We can get rid of this width change by disabling `itemHorizontalTranslation`.
```java
setItemHorizontalTranslationEnabled(false);
```
We only want to disable this if label visibility is "selected" and there are <= 3 tabs. The translation runs an animation when the tab is selected if there are more than 3 tabs, and we don't want to lose that. So you can add this to `TabBar.tsx`
```jsx
<TabView
  labelVisibility={labelVisibility}
  itemHorizontalTranslation={labelVisibility !== 'selected' || tabBarItems.length > 3}
```
Then on the java side you can add a corresponding `setItemHorizontalTranslation` method.

The good news is that this means we can simplify the `setLabelVisibilityMode`. We don't need the `labelVisibility` or `canSetVisibility` fields anymore because, when `itemHorizontalTranslation` is disabled, it doesn't [throw the NullPointer](https://github.com/grahammendick/navigation/issues/577#issuecomment-1034849460) anymore! 
```java
@ReactProp(name = "labelVisibility")
public void setLabelVisibilityMode(TabNavigationView view, int labelVisibilityMode) {
  view.setLabelVisibilityMode(labelVisibilityMode);
}
```

Reply:
Unless... What if we just default to 'auto' when there are more than 3 tabs? It would have the same behaviour, without the need for the java workaround.

Reply:
Yes, that's a great idea! Default to 'auto' when there are more than 3